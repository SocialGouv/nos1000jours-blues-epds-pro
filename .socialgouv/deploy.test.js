const child_process = require("child_process");
const { directory } = require("tempy");
const util = require("util");

const TIMEOUT = 1000 * 60;

const exec = util.promisify(child_process.exec);

const GH_DEFAULTS = {
  GITHUB_JOB: "5678",
  GITHUB_REF: "refs/heads/mybranch",
  GITHUB_REPOSITORY: `socialgouv/nos1000jours-web-pro`,
  GITHUB_RUN_ID: "1234",
  GITHUB_SHA: "0123456789abcdefghijklmnopqrstuvwxyz0123",
  SOCIALGOUV_BASE_DOMAIN: "dev2.fabrique.social.gouv.fr",
  SOCIALGOUV_CONFIG_PATH: __dirname + "/../.socialgouv/config.json",
  SOCIALGOUV_PREPRODUCTION: "",
  SOCIALGOUV_PRODUCTION: "",
};

const degitCommand = ({ dir, env }) => `
  npx degit SocialGouv/kosko-charts/templates/autodevops ${dir}/autodevops; \
  yarn --cwd ${dir}/autodevops --silent; \
  cp -r ${__dirname}/../.socialgouv/environments ${dir}/autodevops/; \
  yarn --cwd ${dir}/autodevops --silent generate --env ${env}
`;

const getEnvironmentManifests = async (envName, defaultEnv = {}) => {
  const dir = directory();

  const env = {
    ...GH_DEFAULTS,
    ...defaultEnv,
  };

  Object.assign(process.env, env);

  const cmd = degitCommand({ dir, env: envName });
  console.info(`Generating ${envName} manifests with autodevops template...`);

  const { stdout: manifests } = await exec(cmd, { env: process.env });

  return manifests;
};

test(
  "generate development k8s manifests",
  async () => {
    const manifests = await getEnvironmentManifests("dev");
    expect(manifests).toMatchSnapshot();
  },
  TIMEOUT
);

test(
  "generate preproduction k8s manifests",
  async () => {
    const manifests = await getEnvironmentManifests("preprod", {
      GITHUB_REF: "refs/tags/v1.3.22",
      SOCIALGOUV_PREPRODUCTION: "true",
    });
    expect(manifests).toMatchSnapshot();
  },
  TIMEOUT
);

test(
  "generate production k8s manifests",
  async () => {
    const manifests = await getEnvironmentManifests("prod", {
      GITHUB_REF: "refs/tags/v1.3.22",
      SOCIALGOUV_PRODUCTION: "true",
      SOCIALGOUV_BASE_DOMAIN: "fabrique.social.gouv.fr",
    });
    expect(manifests).toMatchSnapshot();
  },
  TIMEOUT
);
