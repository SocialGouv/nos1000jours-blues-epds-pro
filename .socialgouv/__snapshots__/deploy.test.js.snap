// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate development k8s manifests 1`] = `
"---
apiVersion: v1
kind: Namespace
metadata:
  annotations:
    socialgouv/creator: autodevops
    janitor/ttl: 15d
    field.cattle.io/creatorId: gitlab
    field.cattle.io/projectId: ''
    git/branch: refs/heads/mybranch
    git/remote: socialgouv/nos1000jours-web-pro
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/nos1000jours-web-pro
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    azure-pg-admin-user: nos1000jours-web-pro
    application: nos1000jours-web-pro
    owner: nos1000jours-web-pro
    team: nos1000jours-web-pro
    cert: wildcard
  name: nos1000jours-web-pro-mybranch
---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/nos1000jours-web-pro
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: nos1000jours-web-pro
    application: nos1000jours-web-pro
    owner: nos1000jours-web-pro
    team: nos1000jours-web-pro
    cert: wildcard
    component: nginx
  name: nos1000jours-web-pro
  namespace: nos1000jours-web-pro-mybranch
spec:
  replicas: 1
  selector:
    matchLabels:
      app: nos1000jours-web-pro
  template:
    metadata:
      annotations:
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        app.github.com/job: '5678'
        app.github.com/ref: refs/heads/mybranch
        app.github.com/repo: socialgouv/nos1000jours-web-pro
        app.github.com/run: '1234'
        app.github.com/sha: '0123456'
      labels:
        app: nos1000jours-web-pro
        application: nos1000jours-web-pro
        owner: nos1000jours-web-pro
        team: nos1000jours-web-pro
        cert: wildcard
        component: nginx
    spec:
      containers:
        - image: >-
            ghcr.io/socialgouv/nos1000jours/nos1000jours-web-pro:sha-0123456789abcdefghijklmnopqrstuvwxyz0123
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /index.html
              port: http
            initialDelaySeconds: 30
            periodSeconds: 5
            timeoutSeconds: 5
          name: nos1000jours-web-pro
          ports:
            - containerPort: 80
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /index.html
              port: http
            initialDelaySeconds: 0
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 1
          resources:
            limits:
              cpu: 500m
              memory: 128Mi
            requests:
              cpu: 5m
              memory: 32Mi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /index.html
              port: http
            periodSeconds: 5
          envFrom:
            - configMapRef:
                name: nos1000jours-web-pro-configmap
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: nos1000jours-web-pro-configmap
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/nos1000jours-web-pro
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    application: nos1000jours-web-pro
    owner: nos1000jours-web-pro
    team: nos1000jours-web-pro
    cert: wildcard
  namespace: nos1000jours-web-pro-mybranch
data:
  NODE_ENV: production
  MATOMO_SITE_ID: '47'
  MATOMO_URL: https://matomo.fabrique.social.gouv.fr/
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: nos1000jours-web-pro
    application: nos1000jours-web-pro
    owner: nos1000jours-web-pro
    team: nos1000jours-web-pro
    cert: wildcard
  name: nos1000jours-web-pro
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/nos1000jours-web-pro
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  namespace: nos1000jours-web-pro-mybranch
spec:
  ports:
    - name: http
      port: 80
      targetPort: 80
  selector:
    app: nos1000jours-web-pro
  type: ClusterIP
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/nos1000jours-web-pro
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: nos1000jours-web-pro
    application: nos1000jours-web-pro
    owner: nos1000jours-web-pro
    team: nos1000jours-web-pro
    cert: wildcard
  name: nos1000jours-web-pro
  namespace: nos1000jours-web-pro-mybranch
spec:
  rules:
    - host: nos1000jours-web-pro-mybranch.dev2.fabrique.social.gouv.fr
      http:
        paths:
          - backend:
              service:
                name: nos1000jours-web-pro
                port:
                  name: http
            path: /
            pathType: Prefix
  tls:
    - hosts:
        - nos1000jours-web-pro-mybranch.dev2.fabrique.social.gouv.fr
      secretName: wildcard-crt
"
`;

exports[`generate preproduction k8s manifests 1`] = `
"---
apiVersion: v1
kind: Namespace
metadata:
  annotations:
    socialgouv/creator: autodevops
    field.cattle.io/creatorId: gitlab
    field.cattle.io/projectId: ''
    git/branch: refs/tags/v1.3.22
    git/remote: socialgouv/nos1000jours-web-pro
    app.github.com/job: '5678'
    app.github.com/ref: refs/tags/v1.3.22
    app.github.com/repo: socialgouv/nos1000jours-web-pro
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    azure-pg-admin-user: nos1000jours-web-pro
    application: nos1000jours-web-pro
    owner: nos1000jours-web-pro
    team: nos1000jours-web-pro
    cert: wildcard
  name: nos1000jours-web-pro-preprod
---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/tags/v1.3.22
    app.github.com/repo: socialgouv/nos1000jours-web-pro
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: nos1000jours-web-pro
    application: nos1000jours-web-pro
    owner: nos1000jours-web-pro
    team: nos1000jours-web-pro
    cert: wildcard
    component: nginx
  name: nos1000jours-web-pro
  namespace: nos1000jours-web-pro-preprod
spec:
  replicas: 1
  selector:
    matchLabels:
      app: nos1000jours-web-pro
  template:
    metadata:
      annotations:
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        app.github.com/job: '5678'
        app.github.com/ref: refs/tags/v1.3.22
        app.github.com/repo: socialgouv/nos1000jours-web-pro
        app.github.com/run: '1234'
        app.github.com/sha: '0123456'
      labels:
        app: nos1000jours-web-pro
        application: nos1000jours-web-pro
        owner: nos1000jours-web-pro
        team: nos1000jours-web-pro
        cert: wildcard
        component: nginx
    spec:
      containers:
        - image: ghcr.io/socialgouv/nos1000jours/nos1000jours-web-pro:1.3.22
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /index.html
              port: http
            initialDelaySeconds: 30
            periodSeconds: 5
            timeoutSeconds: 5
          name: nos1000jours-web-pro
          ports:
            - containerPort: 80
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /index.html
              port: http
            initialDelaySeconds: 0
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 1
          resources:
            limits:
              cpu: 500m
              memory: 128Mi
            requests:
              cpu: 5m
              memory: 32Mi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /index.html
              port: http
            periodSeconds: 5
          envFrom:
            - configMapRef:
                name: nos1000jours-web-pro-configmap
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: nos1000jours-web-pro-configmap
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/tags/v1.3.22
    app.github.com/repo: socialgouv/nos1000jours-web-pro
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    application: nos1000jours-web-pro
    owner: nos1000jours-web-pro
    team: nos1000jours-web-pro
    cert: wildcard
  namespace: nos1000jours-web-pro-preprod
data:
  NODE_ENV: production
  MATOMO_SITE_ID: '47'
  MATOMO_URL: https://matomo.fabrique.social.gouv.fr/
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: nos1000jours-web-pro
    application: nos1000jours-web-pro
    owner: nos1000jours-web-pro
    team: nos1000jours-web-pro
    cert: wildcard
  name: nos1000jours-web-pro
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/tags/v1.3.22
    app.github.com/repo: socialgouv/nos1000jours-web-pro
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  namespace: nos1000jours-web-pro-preprod
spec:
  ports:
    - name: http
      port: 80
      targetPort: 80
  selector:
    app: nos1000jours-web-pro
  type: ClusterIP
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/tags/v1.3.22
    app.github.com/repo: socialgouv/nos1000jours-web-pro
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: nos1000jours-web-pro
    application: nos1000jours-web-pro
    owner: nos1000jours-web-pro
    team: nos1000jours-web-pro
    cert: wildcard
  name: nos1000jours-web-pro
  namespace: nos1000jours-web-pro-preprod
spec:
  rules:
    - host: nos1000jours-web-pro-preprod.dev2.fabrique.social.gouv.fr
      http:
        paths:
          - backend:
              service:
                name: nos1000jours-web-pro
                port:
                  name: http
            path: /
            pathType: Prefix
  tls:
    - hosts:
        - nos1000jours-web-pro-preprod.dev2.fabrique.social.gouv.fr
      secretName: wildcard-crt
"
`;

exports[`generate production k8s manifests 1`] = `
"---
apiVersion: v1
kind: Namespace
metadata:
  annotations:
    socialgouv/creator: autodevops
    field.cattle.io/creatorId: gitlab
    field.cattle.io/projectId: ''
    git/branch: refs/tags/v1.3.22
    git/remote: socialgouv/nos1000jours-web-pro
    app.github.com/job: '5678'
    app.github.com/ref: refs/tags/v1.3.22
    app.github.com/repo: socialgouv/nos1000jours-web-pro
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    azure-pg-admin-user: nos1000jours-web-pro
    application: nos1000jours-web-pro
    owner: nos1000jours-web-pro
    team: nos1000jours-web-pro
  name: nos1000jours-web-pro
---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/tags/v1.3.22
    app.github.com/repo: socialgouv/nos1000jours-web-pro
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: nos1000jours-web-pro
    application: nos1000jours-web-pro
    owner: nos1000jours-web-pro
    team: nos1000jours-web-pro
    component: nginx
  name: nos1000jours-web-pro
  namespace: nos1000jours-web-pro
spec:
  replicas: 1
  selector:
    matchLabels:
      app: nos1000jours-web-pro
  template:
    metadata:
      annotations:
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        app.github.com/job: '5678'
        app.github.com/ref: refs/tags/v1.3.22
        app.github.com/repo: socialgouv/nos1000jours-web-pro
        app.github.com/run: '1234'
        app.github.com/sha: '0123456'
      labels:
        app: nos1000jours-web-pro
        application: nos1000jours-web-pro
        owner: nos1000jours-web-pro
        team: nos1000jours-web-pro
        component: nginx
    spec:
      containers:
        - image: ghcr.io/socialgouv/nos1000jours/nos1000jours-web-pro:1.3.22
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /index.html
              port: http
            initialDelaySeconds: 30
            periodSeconds: 5
            timeoutSeconds: 5
          name: nos1000jours-web-pro
          ports:
            - containerPort: 80
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /index.html
              port: http
            initialDelaySeconds: 0
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 1
          resources:
            limits:
              cpu: 500m
              memory: 128Mi
            requests:
              cpu: 5m
              memory: 32Mi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /index.html
              port: http
            periodSeconds: 5
          envFrom:
            - configMapRef:
                name: nos1000jours-web-pro-configmap
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: nos1000jours-web-pro-configmap
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/tags/v1.3.22
    app.github.com/repo: socialgouv/nos1000jours-web-pro
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    application: nos1000jours-web-pro
    owner: nos1000jours-web-pro
    team: nos1000jours-web-pro
  namespace: nos1000jours-web-pro
data:
  NODE_ENV: production
  MATOMO_SITE_ID: '47'
  MATOMO_URL: https://matomo.fabrique.social.gouv.fr/
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: nos1000jours-web-pro
    application: nos1000jours-web-pro
    owner: nos1000jours-web-pro
    team: nos1000jours-web-pro
  name: nos1000jours-web-pro
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/tags/v1.3.22
    app.github.com/repo: socialgouv/nos1000jours-web-pro
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  namespace: nos1000jours-web-pro
spec:
  ports:
    - name: http
      port: 80
      targetPort: 80
  selector:
    app: nos1000jours-web-pro
  type: ClusterIP
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    certmanager.k8s.io/cluster-issuer: letsencrypt-prod
    kubernetes.io/tls-acme: 'true'
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/tags/v1.3.22
    app.github.com/repo: socialgouv/nos1000jours-web-pro
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: nos1000jours-web-pro
    application: nos1000jours-web-pro
    owner: nos1000jours-web-pro
    team: nos1000jours-web-pro
  name: nos1000jours-web-pro
  namespace: nos1000jours-web-pro
spec:
  rules:
    - host: nos1000jours-web-pro.fabrique.social.gouv.fr
      http:
        paths:
          - backend:
              service:
                name: nos1000jours-web-pro
                port:
                  name: http
            path: /
            pathType: Prefix
  tls:
    - hosts:
        - nos1000jours-web-pro.fabrique.social.gouv.fr
      secretName: nos1000jours-web-pro-crt
"
`;
